tests/Browser/LoginTest.php

```diff
@@ -0,0 +1,236 @@
+     * Click forget passwork link
+     * Case 1-4
+     */
+    public function testForgetPasswordLink()
+    {
+        $data = $this->getDatatTest();
+        $this->browse(function (Browser $browser) use ($data) {
+            $browser->visit($data['path']['login'])
+                    ->clickLink('Forgot Your Password?')
+                    ->assertPathIs('/password/reset');
+        });
+    }
+
+    /**
+     * Test path register
+     * Case 1-5
+     */
+    public function testRegisterLink()
+    {
+        $data = $this->getDatatTest();
+        $this->browse(function (Browser $browser) use ($data) {
+            $browser->visit($data['path']['login'])
+                    ->clickLink('Register')
+                    ->assertPathIs('/register');
+        });
+    }
+
+    /**
+     * Login with email or password wrong
+     * Case 1-6
+     */
+    public function testLoginNotInput()
+    {
+        $data = $this->getDatatTest();
+        $this->browse(function ($browser) use ($data) {
+            $browser->visit($data['path']['login'])
+                    ->press('Login')
+                    ->assertFocused('email');
+        });
+    }
+
+    /**
+     * Login with email or password wrong keep email login
+     * Case 1-7
+     */
+    public function testUserWrong()
+    {
+        $data = $this->getDatatTest();
+        $this->browse(function ($browser) use ($data) {
+            $browser->visit($data['path']['login'])
+                    ->type('email', $data['email']['user'])
+                    ->type('password', 'abc@123')
+                    ->press('Login')
+                    ->assertSee('These credentials do not match our records.')
+                    ->assertInputValue('email', $data['email']['user']);
+        });
+    }
+
+    /**
+     * User login with email, password
+     * Case 2-1
+     */
+    public function testUserLogin()
+    {
+        $data = $this->getDatatTest();
+        $this->browse(function ($browser) use ($data) {
+            $browser->visit($data['path']['login'])
+                    ->type('email', $data['email']['user'])
+                    ->type('password', 'lifull@123')
+                    ->press('Login')
+                    ->assertPathIs('/')
+                    ->logout();
+        });
+    }
+
+    /**
+     * User logged in go to path '/admin'
+     * Case 2-2
+     */
+    public function testUserVisitAdmin()
+    {
+        $data = $this->getDatatTest();
+        $this->browse(function ($browser) use ($data) {
+            $browser->loginAs(User::find(2))
+                    ->visit($data['path']['admin'])
+                    ->assertPathIs($data['path']['user'])
+                    ->logout();
+        });
+    }
+
+    /**
+     * Admin login with email, password.
+     * Case 2-3
+     */
+    public function testAdminLogin()
+    {
+        $data = $this->getDatatTest();
+        $this->browse(function ($browser) use ($data) {
+            $browser->logout()
```

Comment:
why keep have logout  begin or end ? if you want reset state , learn to use tearDown()